BHoM_UI/Components/Adapter/Pull.cs

```diff
@@ -80,19 +81,43 @@ public override object Run(object[] inputs)
         /**** Public Method               ****/
         /*************************************/
 
-        [Description("Pull objects from the external software")]
+        [Description("Adapter Action 'Pull': imports objects from an external software. \nAlso allows for querying through Requests.")]
         [Input("adapter", "Adapter to the external software")]
-        [Input("request", "Filter on the objects to pull")]
-        [Input("config", "Pull config")]
+        [Input("request", "Input a Request to pull only objects that satisfy a certain rule. \n" + 
+            "Input a Type to pull only objects of that type.")]
+        [Input("pullType", "Pull type. Connect the enum PullType for all the alternatives.")]
+        [Input("actionConfig", "Configuration for this Action. You can input an ActionConfig (it contains the configs common to all Toolkits); \n" +
+            "consider that Toolkits may have a custom ActionConfig (e.g. GSAConfig, SpeckleConfig).")]
         [Input("active", "Execute the pull")]
-        [Output("objects","Objects pulled")]
-        public static IEnumerable<object> Pull(BHoMAdapter adapter, IRequest request = null, Dictionary<string, object> config = null, bool active = false)
+        [Output("objects", "Objects pulled")]
+        public static IEnumerable<object> Pull(BHoMAdapter adapter, object request = null,
+            PullType pullType = PullType.AdapterDefault, ActionConfig actionConfig = null,
+            bool active = false)
         {
+            // ---------------------------------------------//
+            // Mandatory Adapter Action set-up              //
+            //----------------------------------------------//
+            // The following are mandatory set-ups to be ALWAYS performed 
+            // before the Adapter Action is called,
+            // whether the Action is overrided at the Toolkit level or not.
+
+            // If unset, set the actionConfig to a new ActionConfig.
+            actionConfig = actionConfig == null ? new ActionConfig() : actionConfig;
+
+            // Always assure there is a Request. Allow to input a Type to generate a FilterRequest.
+            IRequest actualRequest = null;
+
             if (request == null)
-                request = new FilterRequest();
+                actualRequest = new FilterRequest();
+            else if (request is Type)
+                actualRequest = BH.Engine.Data.Create.FilterRequest((Type)request, "");
+            else if (typeof(IRequest).IsAssignableFrom(request.GetType()))
+                actualRequest = request as IRequest;
```

Comment:
Just to reinforce, this code is here because we allow the override of the `Pull` method and. because of that, we are not sure the `base.Pull` will be called - thus, we cannot enforce this logic.

The more I read the code, the more I think that the `Pull` method should not be `virtual`, and anything the user wants to do, can be done in a `BeforePull` and `AfterPull` methods.